tests/unit_tests/tools/skill_tools/test_sql_summary_skill.py

```
@@ -0,0 +1,295 @@
+            text=True,
+            timeout=30,
+        )
+        assert result.returncode == 0
+        assert result.stdout.strip()  # Non-empty ID
+
+    def test_generate_id_deterministic(self, sql_summary_skill_dir):
+        """Same SQL produces same ID."""
+        sql = "SELECT user_id, COUNT(*) FROM orders GROUP BY user_id"
+        results = []
+        for _ in range(2):
+            result = subprocess.run(
```

Ruff (0.15.5)

[error] `subprocess` call: check for execution of untrusted input

(S603)